impl/pkg/dht/dht.go

```diff
@@ -29,12 +32,40 @@ func NewDHT(bootstrapPeers []string) (*DHT, error) {
 	return &DHT{Server: s}, nil
 }
 
+// NewTestDHT returns a new instance of DHT that does not make external connections
+func NewTestDHT(t *testing.T, bootstrapPeers ...dht.Addr) *DHT {
```

Comment:
I don't love having test code in an exported package, especially since the DHT itself is not broadly useful to consumers of this code...but I can understand why this decision was made - avoid needing to drag DHT logic into the place where it's tested, and go is annoying about not re-using test utilities across packages.

Suffice to say - don't love it, get it, wonder if there's something better, but if not this is fine.

Reply:
That roughly lines up with my thoughts. I tried putting it in the `test_dht` package, but I couldn't import it from other packages. I will spend a few minutes looking for better options. Surely this is something that others have run into.

Reply:
Some research points me to [this reddit thread](https://old.reddit.com/r/golang/comments/jjmsoc/how_to_share_test_utilities_accros_packages/) which proposes the idea of having a the exported test function in a file with `package dht` (ie. not `dht_test`) but filename ending in `_test.go`. However, when I tried that other test files were still unable to import it. I'll leave it here for now and if it becomes a problem or someone comes up with a better idea we can address it further.